File: app-android/app/src/main/java/de/tutao/tutanota/EncryptionUtils.java
package de.tutao.tutanota;

import java.nio.charset.StandardCharsets;
import java.util.Date;

public class EncryptionUtils {
	public static Date decryptDate(String encryptedData, Crypto crypto, byte[] sessionKey) throws CryptoError {
		byte[] decBytes = crypto.aesDecrypt(sessionKey, encryptedData);
		return new Date(Long.valueOf(new String(decBytes, StandardCharsets.UTF_8)));
	}

	public static String decryptString(String encryptedData, Crypto crypto, byte[] sessionKey) throws CryptoError {
		byte[] decBytes = crypto.aesDecrypt(sessionKey, encryptedData);
		return new String(decBytes, StandardCharsets.UTF_8);
	}

	public static long decryptNumber(String encryptedData, Crypto crypto, byte[] sessionKey) throws CryptoError {
		String stringValue = decryptString(encryptedData, crypto, sessionKey);
		return Long.parseLong(stringValue);
	}
}

File: app-android/app/src/main/java/de/tutao/tutanota/EncryptionUtils.kt
package de.tutao.tutanota

import java.nio.charset.StandardCharsets
import java.util.*

@Throws(CryptoError::class)
fun AndroidNativeCryptoFacade.decryptDate(encryptedData: String, sessionKey: ByteArray): Date {
	val decBytes = aesDecrypt(sessionKey, encryptedData)
	return Date(String(decBytes, StandardCharsets.UTF_8).toLong())
}

@Throws(CryptoError::class)
fun AndroidNativeCryptoFacade.decryptString(encryptedData: String, sessionKey: ByteArray): String {
	val decBytes = aesDecrypt(sessionKey, encryptedData)
	return String(decBytes, StandardCharsets.UTF_8)
}

@Throws(CryptoError::class)
fun AndroidNativeCryptoFacade.decryptNumber(encryptedData: String, sessionKey: ByteArray): Long {
	val stringValue = decryptString(encryptedData, sessionKey)
	return stringValue.toLong()
}